sound/soc/sof/intel/hda-loader.c

```diff
@@ -325,7 +325,7 @@ int hda_dsp_cl_boot_firmware_iccmax(struct snd_sof_dev *sdev)
 	iccmax_stream = get_stream_with_tag(sdev, iccmax_tag, SNDRV_PCM_STREAM_CAPTURE);
 	if (!iccmax_stream) {
 		dev_err(sdev->dev, "error: could not get stream with stream tag %d\n", iccmax_tag);
-		ret = -ENODEV;
+		return -ENODEV;
```

Comment:
Are we 100% certain that there is no cleanup necessary if iccmax_stream is null? I'm a little bit concerned about a potential memory leak (for example, shouldn't cl_cleanup run anyway and handle this case specially?)

Reply:
yeah, I don't get this either and did ask the same question earlier. @ranj063 this one is for you

Reply:
This is valid issue but likely need a slightly different fix. If tag  were invalid, we'd never get to get_stream_with_tag(). So, in reality we'd never fail at this step. Let me look into fixing it differently.

Reply:
I will close this PR. Thanks @ranj063 for look into the issue.